documentation/pwa/tutorial-pwa-offline.asciidoc

```diff
@@ -1,23 +1,88 @@
 ---
-title: PWA Offline Page
+title: PWA Offline Experience
 order: 6
 layout: page
 ---
 
-= PWA Offline Page
+= PWA Offline Experience
 
-Vaadin automatically generates and serves an offline page. This is a simple page that:
+Vaadin supports two alternative ways of building offline experience:
 
-* Includes the application name and icon.
-* Communicates to the user that the application is offline, because there is no network connection.
+* Client-side TypeScript views (default)
+* A separate offline page
+
+For PWAs built with Vaadin, the service worker provides offline support for TypeScript routes and views. By default, it stores the application shell HTML, the compiled frontend bundles, and the other necessary resources, and then serves them offline from the browser’s cache.
+
+You can optionally use a separate offline page instead of TypeScript views offline (`offlinePath` property in `@PWA` annotation).
```

Comment:
As an uninformed developer reading this, I would expect some kind of nudge towards which option (TS view or `offlinePath`) I should go with based on what I want to do. I.e. if I'm about to build something that has some kind of custom view logic in offline mode I should probably go with TS views, otherwise if I mostly want to display static resoucs maybe `offlinePath` is sufficient.